docs/DeveloperGuide.md

```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+**MSS**
+1. User requests to list all services that matches the search value.
+1. GrAB3 searches for services that matches search value.
+1. GrAB3 show list of all services that matches the search value.
+Use case ends.
+
+**Extensions**
+* 2a. No service matches the search value.
+  * 2a1. GrAB3 displays empty list.
+  Use case resumes at step 1.
+  
+#### UC005: List Services 
+List all the services in the service list.
+
+**Preconditions: Service exists in service list.**
+ 
+**Guarantees: All services in the service list will be displayed.**
+ 
+**MSS**
+1. User requests to list all services.
+1. GrAB3 display all services in the service list.
+Use case ends.
+
+**Extensions**
+* 1a. No service in service list.
+  * 1a1. GrAB3 displays empty list.
+  Use case resumes at step 1.
+  
+#### UC005: Clear Services
+Clear all services in the service list.
+
+**Preconditions: Service exists in service list**
+
+**Guarantees: An empty revenue list.**
+
+**MSS**
+1. User requests to clear the list.
+1. GrAB3 shows the updated list.
+Use case ends.
 
 ### Client List
 
-**UC02: Add a client to client list**
+#### UC001: Add a client to client list
 
 **Preconditions: A client must exist**
 
 **Guarantees: A new client will added to the client list** 
 
 **MSS**
-
 1.  User requests to add a new client into the client list.
-2.  GrAB3 shows the updated client list
-
-    Use case ends.
+1.  GrAB3 shows the updated client list
+Use case ends.
 
 **Extensions**
 
 * 1a. Client description is empty.
-
     * 1a1. GrAB3 shows an error message.
-    
-        Use case resumes at step 2.
+    Use case resumes at step 2.
 
-**UC03: Delete a client**
+#### UC003: Delete a client**
 
 **Preconditions: client schedule must exist**
 
 **Guarantees: The client specified by the user will be deleted**
 
 **MSS**
-
 1.  User requests to list all clients.
-2.  GrAB3 displays a list of clients.
-3.  User requests to delete a specific client from the list generated.
-4.  GrAB3 shows the updated list after deletion of the client.
-
-    Use case ends.
+1.  GrAB3 displays a list of clients.
+1.  User requests to delete a specific client from the list generated.
+1.  GrAB3 shows the updated list after deletion of the client.
+Use case ends.
 
 **Extensions**
-
 * 3a. The specified client does not exist.
-
     * 3a1. GrAB3 shows an error message.
+    Use case resumes at step 2.
 
-      Use case resumes at step 2.
-
-**UC04: Edit a client**
+#### UC004: Edit a client
 
 **Preconditions: Client exists in TaskSchedule**
 
 **Guarantees: Client specified by user gets edited**
 
 **MSS**
-
 1.  User requests to list all clients.
-2.  GrAB3 shows a list of clients.
-3.  User requests to edit the description of a specific client from that list.
-4.  GrAB3 shows the updated client.
-
-    Use case ends.
+1.  GrAB3 shows a list of clients.
+1.  User requests to edit the description of a specific client from that list.
```

Comment:
maybe can change description to details?

Reply:
okay